src/Sarif.UnitTests/Visitors/ValidatingVisitorTests.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.IO;
+using FluentAssertions;
+using System.Text;
+using Newtonsoft.Json;
+using Microsoft.CodeAnalysis.Sarif.Readers;
+
+namespace Microsoft.CodeAnalysis.Sarif.Visitors
+{
+    [TestClass]
+    public class ValidatingVisitorTests
+    {
+        [TestMethod]
+        public void ValidatingVisitor_InvalidUri()
+        {
+            // NOTE that for validating tests, we need to deserialize SARIF text 
+            // without benefit of the many converters that transform specified
+            // JSON values to enum equivalents, etc. The Sarif text below and 
+            // for all these tests is rendered, in some cases, as it the converter
+            // had done some processing before conversion to eventual type. We
+            // express the version as OneZeroZeroBeta5, for example, because the
+            // default JSON.NET enum conversion will do the right thing as far
+            // as converting this value to the C# OM.
```

Comment:
Ohh! Lightbulb. I had to read this several times before I understood. You want to examine the result of deserializing a SARIF file without the benefit of any of our converters that might "help out" and mask problems in the raw file. But if (for example) sarifLog.version had a "real" value like "1.0.0", Json.NET couldn't deserialize it into a property whose type is SarifVersion. So you fake up a file that *can* be deserialized without benefit of our converters.

Ok, I get that, but how will this work in real life? A hypothetical command line tool that performs this kind of detailed validation is going to be deserializing real files with values like "version": "1.0.0". So the tool will need to use a contract resolver that does at least some of what the "real" resolver does, like convert enums.

So there are two choices. Either the tool uses the "real" resolver, and maybe misses a chance to point out some problems, or it creates its own resolver that uses only a subset of the converters used by the real resolver -- like maybe only the enum converter.

Either way, the tests ought to be doing what a "real" validation tool would do.